core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSource.scala

```diff
@@ -107,6 +108,11 @@ trait DeltaSourceBase extends Source
 
   protected var lastOffsetForTriggerAvailableNow: DeltaSourceOffset = _
 
+  override def prepareForTriggerAvailableNow(): Unit = {
+    lastOffsetForTriggerAvailableNow = DeltaSourceOffset(deltaLog.tableId,
+      latestOffset(null, ReadLimit.allAvailable()))
```

Comment:
IIRC, when the table contains no file, `latestOffset` will return `null`. Could you add a test to make sure we won't hit NPE?

Reply:
Wow good call. It does indeed throw an NPE. Fixed and added a test